### PR TITLE
LLM: Disable transformer api `pretraining_tp`

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

To avoid linear parallel involved F.linear operation, we turn off `pretraining_tp` in large models.
https://github.com/analytics-zoo/nano/issues/482

### CODE REVIEW

**Review by yangw1234 — APPROVED** [review]

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 1]
```diff
@@ -72,6 +72,9 @@ def from_pretrained(cls,
             kwargs["low_cpu_mem_usage"] = True
             # set default torch_dtype='auto'
             kwargs["torch_dtype"] = kwargs.get("torch_dtype", 'auto')
+            # Avoid tensor parallel F.Linear Operations
+            if "pretraining_tp" in config_dict:
+                kwargs["pretraining_tp"] = 0
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/model.py:77]: should it be 1?

Comment (leonardozcm) [reply python/llm/src/bigdl/llm/transformers/model.py:77]: I think both 0 and 1 will prevent this problem, only if self.pretraining_tp <= 1
```
        if self.pretraining_tp > 1: 
```

Comment (jason-dai) [reply python/llm/src/bigdl/llm/transformers/model.py:77]: what the default value for 7B/13B?

Comment (leonardozcm) [reply python/llm/src/bigdl/llm/transformers/model.py:77]: They don't have pretraining_tp this entry in their config.json

**Comment by jason-dai** [comment]
https://huggingface.co/docs/transformers/main/model_doc/llama2
```
Setting config.pretraining_tp to a value different than 1 will activate the more accurate but slower computation of the linear layers, which should better match the original logits.
```

**Comment by leonardozcm** [comment]
OK I have set this to default 1 as a precaution

**Pull request merged**